codegen/grid.py

```diff
@@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
             var = var.name
         self.defined_variable[var] = Variable(var, value, type, constant)
 
+    def set_page_size(self, page_size):
```

Comment:
Why are you using a global variable? When generating code can’t you directly write the alignment size into the expression rather than using a C variable?

Also note that you’ll have to compile with -std=c++x11

On 21 Aug 2015, at 11:23, tj-sun <notifications@github.com<mailto:notifications@github.com>> wrote:


In codegen/grid.py<https://github.com/opesci/propagator/pull/25#discussion_r37622578>:

> @@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
>              var = var.name
>          self.defined_variable[var] = Variable(var, value, type, constant)
>
> +    def set_page_size(self, page_size):


  *   function name changed to set_alignment
  *   added parameter int size to int opesci_read_simple_binary_ptr(), compare size with file size when reading data
  *   having googled a bit I'm not sure how to allocate dynamic memory using attribute, apart from using global variable. These guys seem to be using posix: http://www.prace-ri.eu/IMG/pdf/Best-Practice-Guide-Intel-Xeon-Phi.pdf

—
Reply to this email directly or view it on GitHub<https://github.com/opesci/propagator/pull/25/files#r37622578>.